File: DotNetflixServer/BackendAPI/Dto/UserDto.cs
using DBModels.IdentityLogic;

namespace BackendAPI.Dto
{
    public class UserDto
    {
        public string Login { get; set; }
        public string Email { get; set; }
        public DateTime Birthdate { get; set; }
        public bool Enabled2FA { get; set; }
        public Role Role { get; set; }
    }
}

File: mobile/lib/models/user_dto.dart
part of 'all_for_freezed.dart';

@freezed
class UserDto with _$UserDto {
  const factory UserDto({
    required String login,
    required String email,
    required DateTime birthdate,
    required bool enabled2FA,
  }) = _UserDto;

  factory UserDto.fromJson(Map<String, dynamic> json) =>
      _$UserDtoFromJson(json);
}
